synfig-studio/src/gui/trees/layertreestore.cpp

```diff
@@ -894,7 +894,15 @@ LayerTreeStore::set_row_layer(Gtk::TreeRow &row, const synfig::Layer::Handle &ha
 			std::set<String> impossible_existant_layers;
 			if (etl::handle<Layer_Switch> layer_switch = etl::handle<Layer_Switch>::cast_dynamic(handle))
 			{
-				if (!layer_switch->get_param("layer_name").get(String()).empty())
+				// is this Switch operated by layer-name or by depth-index?
+				bool does_param_layer_name_change = false;
+				auto iter = layer_switch->dynamic_param_list().find("layer_name");
+				if (iter != layer_switch->dynamic_param_list().end()) {
+					std::set<ValueBase> values;
+					iter->second->get_values(values);
+					does_param_layer_name_change = true;
```

Comment:
This part seems incomplete to me. 
Maybe this is debugging code? If yes, just comment it for future references:
```
// debug values
// std::set<ValueBase> values;
// iter->second->get_values(values);
```

or maybe it should be something like this?
```
std::set<ValueBase> values;
iter->second->get_values(values);
does_param_layer_name_change = values.size() > 1;
```

Reply:
Indeed.
I'll fix in other way, however.
I notice there are useless calls in my code  :) Thank you

Reply:
Done.